### PR TITLE
Update to 2.3.1

### PR DESCRIPTION
**Pull request by jpugliesi**
Upgrades papermill to 2.3.1

This contains a fix to an issue introduced in a change to nbformat ([see this papermill issue](https://github.com/nteract/papermill/issues/568)) that will raise an error like:
```
Notebook JSON is invalid: Additional properties are not allowed ('id' was unexpected)
```

I'm not sure if this was manifested due to this recent nbformat version bump: https://github.com/viaduct-ai/vml-py/pull/355

### CODE REVIEW

**Review by aleksandrpetukhov — APPROVED** [review]

File: papermill/utils.py [hunk 1]
```diff
@@ -10,6 +10,48 @@
...
+
+    Returns
+    -------
+    bool
+        Whether the notebook contains a cell tagged ``tag``?
+    """
+    return any([tag in cell.metadata.tags for cell in nb.cells])
+
+
+def find_first_tagged_cell_index(nb, tag):
```
Comment (devstein) [inline papermill/utils.py:31]: We could probably use these in our [papermill wrapper](https://github.com/viaduct-ai/vml-py/blob/master/src/vml/papermill/papermill_sparkmagic_wrapper.py)

File: papermill/translators.py [hunk 2]
```diff
@@ -377,6 +547,8 @@ def assign(cls, name, str_val):
 papermill_translators.register("matlab", MatlabTranslator)
 papermill_translators.register(".net-csharp", CSharpTranslator)
 papermill_translators.register(".net-fsharp", FSharpTranslator)
+papermill_translators.register(".net-powershell", PowershellTranslator)
+papermill_translators.register("pysparkkernel", PythonTranslator)
```
Comment (devstein) [inline papermill/translators.py:551]: What about the `sparkkernel` and SparkK kerenels?

**Review by devstein — APPROVED** [review]

File: papermill/utils.py (hunk 1 above)
Comment (jpugliesi) [reply papermill/utils.py:31]: truth

File: papermill/translators.py (hunk 2 above)
Comment (jpugliesi) [reply papermill/translators.py:551]: gp - registered

**Pull request merged**